codice/ddf
# [2.16.x] DDF-5501 Added support for upgrading Catalog configurations with migration

#### What does this PR do?
Adds support for upgrading `Catalog Federation Strategy`, `Metacard Validation Filter Plugin`, `Metacard Validation Marker Plugin`, and `Catalog Standard Framework` Catalog configurations with migration across versions.

#### Who is reviewing it? 

@tyler30clemens
@mdang8
@aaronilovici
@josephthweatt
@oconnormi
@austinsteffes
@ahoffer

#### Select relevant component teams: 

@codice/core-apis

#### Ask 2 committers to review/merge the PR and tag them here.

@brjeter
@paouelle

#### How should this be tested?

1. Install an old release version of DDF that is in the list of "supported versions" and create `Catalog Federation Strategy`, `Metacard Validation Filter Plugin`, `Metacard Validation Marker Plugin`, and `Catalog Standard Framework` configurations under the Catalog App
2. Run `migration:export`, which creates 3 files under the `/exported` directory
3. Shutdown
4. Start up a DDF built on this PR's changes
5. Install with `profile:install standard`
6. Copy all 3 files from step 2 into this newly installed DDF under `/exported`.
7. Run `migration:import -f`
8. After the system automatically restarts, verify all config changes are present in the new DDF and they work

#### What are the relevant tickets?
Fixes: #5501 

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
